template-parts/post-list-filters.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Template for post list filter controls
+ *
+ * @package shiro
+ */
+
+// Get query vars.
+$query_var_search_term = get_search_query();
+$query_var_date_from   = isset( $_GET['date_from'] ) ? sanitize_text_field( $_GET['date_from'] ) : '';
+$query_var_date_to     = isset( $_GET['date_to'] ) ? sanitize_text_field( $_GET['date_to'] ) : '';
+$query_var_categories  = isset( $_GET['categories'] ) ? array_map( 'sanitize_text_field', $_GET['categories'] ) : [];
```

Comment:
Thanks @goldenapples , I absorbed your preference as it's a good idea, allow reusing the vars and avoid applying sanitization many times.